src/test/java/com/gildedrose/GildedRoseTest.java

```diff
@@ -2,16 +2,160 @@
 
 import static org.junit.Assert.*;
 
+import org.junit.Before;
 import org.junit.Test;
 
 public class GildedRoseTest {
 
+    private static final String ITEM_DEFAULT_NAME = "foo";
+    private static final int ITEM_DEFAULT_SELLIN = 15;
+    private static final int ITEM_DEFAULT_QUALITY = 15;
+    private static final int ITEM_OVERDUE_SELLIN = 0;
+    private static final int ITEM_LONG_OVERDUE_SELLIN = -50;
+    private static final int ITEM_QUALITY_MAX = 50;
+    private static final String ITEM_NAME_BACKSTAGE_PASS = "Backstage passes to a TAFKAL80ETC concert";
+    private static final String ITEM_NAME_AGED_BRIE = "Aged Brie";
+    private static final String ITEM_NAME_SULFURAS = "Sulfuras, Hand of Ragnaros";
+    private static final int ITEM_QUALITY_MIN = 0;
+    private Item item;
+    private GildedRose app;
+
+    @Before
+    public void setUp() {
+        item = new Item(ITEM_DEFAULT_NAME, ITEM_DEFAULT_SELLIN, ITEM_DEFAULT_QUALITY);
+        final Item[] items = new Item[] {item};
+        app = new GildedRose(items);
+    }
+
+    @Test
+    public void doesNotAlterName() {
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_NAME, app.items[0].name);
+    }
+
+    @Test
+    public void doesDecreaseSellIn() {
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_SELLIN - 1, app.items[0].sellIn);
+    }
+
+    @Test
+    public void doesDecreaseQuality() {
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_QUALITY - 1, app.items[0].quality);
+    }
+
+    @Test
+    public void withPassedSellIn_doesDecreaseQuality() {
+        item.sellIn = ITEM_OVERDUE_SELLIN;
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_QUALITY - 2, app.items[0].quality);
+    }
+
+    @Test
+    public void doesNotDecreaseQualityBelowZero() {
+        item.quality = ITEM_QUALITY_MIN;
+        app.updateQuality();
+        assertEquals(ITEM_QUALITY_MIN, app.items[0].quality);
+    }
+
+    @Test
+    public void withNegativeQuality_doesNotDecreaseQuality() {
+        item.quality = -1;
+        app.updateQuality();
+        assertEquals(-1, app.items[0].quality);
+    }
+
+    @Test
+    public void withAgedBrie_doesIncreaseQuality() {
+        item.name = ITEM_NAME_AGED_BRIE;
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_QUALITY + 1, app.items[0].quality);
+    }
+
+    @Test
+    public void withAgedBrieAfterSellIn_doesIncreaseQualityTwice() {
+        item.name = ITEM_NAME_AGED_BRIE;
+        item.sellIn = ITEM_OVERDUE_SELLIN;
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_QUALITY + 2, app.items[0].quality);
```

Comment:
as the test method name states `Twice`, i'd expect the common increase value multiplied by two. extracting a constant of value 1 which defines the "growth speed", which can be doubled afterwards.